src/libs/import.c

```diff
@@ -1511,6 +1534,19 @@ static void _lib_import_select_folder(GtkWidget *widget, dt_lib_module_t *self)
 {
   dt_lib_import_t *d = (dt_lib_import_t *)self->data;
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
+
+#ifdef WIN32
+  GtkFileChooserNative *filechooser = gtk_file_chooser_native_new(
+      _("open folder"), GTK_WINDOW(win), GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER, _("_open"), _("_cancel"));
+  // run the native dialog
+  if(gtk_native_dialog_run(GTK_NATIVE_DIALOG(filechooser)) == GTK_RESPONSE_ACCEPT)
+  {
+    char *dirname = gtk_file_chooser_get_filename(GTK_FILE_CHOOSER(filechooser));
+    _add_custom_place(dirname, self);
+    g_free(dirname);
+  }
+  g_object_unref(filechooser);
+#else
```

Comment:
Instead of wrapping that in ifdef, replace normal filechooser with native filechooser. That should work nicely on both windows and mac AND be indistinguishable on linux. Also using native filechooser is recommended inside flatpacked apps :)

Reply:
done, now it need testing on linux too ;-)